src/components/tx/SignOrExecuteForm/hooks.ts

```diff
@@ -135,3 +136,22 @@ export const useIsExecutionLoop = (): boolean => {
   const { safeAddress } = useSafeInfo()
   return wallet ? sameString(wallet.address, safeAddress) : false
 }
+
+// Returns true if the safeTxGas equals the block gas limit
+export const useSafeTxGasError = (safeTx?: SafeTransaction): boolean | undefined => {
+  const [hasSafeTxGasError, setHasSafeTxGasError] = useState<boolean>()
+
+  useEffect(() => {
+    if (!safeTx) return
+
+    const getBlockGasLimit = async () => {
+      const blockGasLimit = await getLatestBlockGasLimit()
+      const hasError = blockGasLimit <= safeTx.data.safeTxGas
+      setHasSafeTxGasError(hasError)
+    }
+
+    getBlockGasLimit()
+  }, [safeTx])
```

Comment:
I'd suggest using `useAsync` for this. We are not checking whether the hook is mounted or if `getLatestBlockGasLimit` throws otherwise.